[RA2 Ch3] HW Topology Management (#2150)
cntt-n/CNTT: doc/ref_arch/kubernetes/chapters/chapter03.md

should be allocated to account for consideration of the underlying HW topology.
This newer feature is missing from Kubernetes, therefore a gap has been
identified and added to [Chapter 6.2.8](./chapter06.md#628-hw-topology-aware-hugepages)

<a name="3.2.1.4"></a>
#### 3.2.1.4 Hardware Topology Management

> Scheduling pods across NUMA boundaries can result in lower performance and higher latencies for applications that require optimizations of CPU isolation, memory and device locality.

Reviewer: I don't think the qualification at the end is necessary
```suggestion
Scheduling pods across NUMA boundaries can result in lower performance and higher latencies for applications.
```

Author: > I don't think the qualification at the end is necessary

@pgoyal01 : How about keeping it in separate sentence? Please check current text.

> Kubernetes supports Topology Policy per node as beta feature ([documentation](https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/) and not per pod. The Topology Manager receives Topology information from Hint Providers which identify NUMA nodes (defined as server system architecture divisions of CPU sockets) and preferred scheduling. In the case of the pod with Guaranteed QoS class having integer CPU request, the static CPU Manager policy would return topology hints relating to the exclusive CPU and the Device Manager would provide hints for the requested device.

Reviewer: nit pick
```suggestion
Kubernetes supports Topology policy per node as beta feature ([documentation](https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/) and not per pod. The Topology Manager receives Topology information from Hint Providers which identify NUMA nodes (defined as server system architecture divisions of CPU sockets) and preferred scheduling. In the case of the pod with Guaranteed QoS class having integer CPU requests, the static CPU Manager policy would return topology hints relating to the exclusive CPU and the Device Manager would provide hints for the requested device.
```

> Memory or Huge Pages are not considered by the Topology Manager. This can be done by the operating system providing best-effort local page allocation for containers (as long as sufficient HugePages are free on the local memory node), or with Control Groups (cgroups) cpuset subsystem that can isolate memory to single NUMA node.

Reviewer: ```suggestion
Memory or Huge Pages are not considered by the Topology Manager. This can be done by the operating system providing best-effort local page allocation for containers as long as there is sufficient free local memory on the node, or with Control Groups (cgroups) cpuset subsystem that can isolate memory to single NUMA node.
```
